### PR TITLE
cleanup

### PR DESCRIPTION
**Pull request by denar90**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -6,30 +6,26 @@ const sysPath = require('path');
 
 class HandlebarsCompiler {
   constructor(cfg) {
-    if (cfg == null) cfg = {};
     this.optimize = cfg.optimize;
     const config = cfg.plugins && cfg.plugins.handlebars || {};
```
Comment (shvaikalesh) [inline index.js]: `cfg.plugins &&` is extra

File: index.js [hunk 2]
```diff
@@ -6,30 +6,26 @@ const sysPath = require('path');
...
     const overrides = config.overrides;
     if (typeof overrides === 'function') overrides(handlebars);
     this.namespace = config.namespace;
     this.pathReplace = config.pathReplace || this.pathReplace;
-    if (config.include) this.includeSettings = config.include;
+    this.include = config.include;
+  }
 
-    this.setInclude();
+  set include(value) {
```
Comment (shvaikalesh) [inline index.js]: setters return value is ignored, so this one does nothing 😄

File: index.js [hunk 3]
```diff
@@ -6,30 +6,26 @@ const sysPath = require('path');
...
 class HandlebarsCompiler {
   constructor(cfg) {
-    if (cfg == null) cfg = {};
     this.optimize = cfg.optimize;
     const config = cfg.plugins && cfg.plugins.handlebars || {};
     const overrides = config.overrides;
     if (typeof overrides === 'function') overrides(handlebars);
     this.namespace = config.namespace;
     this.pathReplace = config.pathReplace || this.pathReplace;
-    if (config.include) this.includeSettings = config.include;
```
Comment (shvaikalesh) [inline index.js]: `this.includeSettings = config.include || {}` and use it in getter

File: index.js [hunk 4]
```diff
@@ -6,30 +6,26 @@ const sysPath = require('path');
...
-    const include = this.includeSettings || {};
-    if (include.enabled === false) {
-      delete HandlebarsCompiler.prototype.include;
-      return;
-    }
+  get include() {
     let includeFile = 'handlebars';
-    if (include.runtime || include.runtime == null) includeFile += '.runtime';
-    if (include.amd) includeFile += '.amd';
+    if (this.include.runtime || this.include.runtime == null) includeFile += '.runtime';
```
Comment (shvaikalesh) [inline index.js]: you can `include` getter within include getter -- would result in `RangeError`

File: index.js [hunk 5]
```diff
@@ -47,13 +43,7 @@ class HandlebarsCompiler {
...
     try {
       const source = `Handlebars.template(${handlebars.precompile(data)})`;
-
-      let ns;
-      if (typeof this.namespace === 'function') {
-        ns = this.namespace(path);
-      } else {
-        ns = this.namespace;
-      }
+      const ns = (typeof this.namespace === 'function') ? this.namespace(path) : this.namespace;
```
Comment (shvaikalesh) [inline index.js]: I think `this.namespace` can be normalized to function in `constructor`

Comment (denar90) [inline index.js]: But we don't know `path` value in constructor, do we?

Comment (shvaikalesh) [inline index.js]: Right, `const ns = this.namespace(path)` and

```js
this.namespace = typeof stuff === 'function' ? stuff : path => path
```

in `constructor`

Comment (denar90) [inline index.js]: make sense

**Pull request merged**